### DISCUSSION TITLE
Can model compression be used for GAN？

### DISCUSSION THREAD

**Discussion by yy2016417** [microsoft/nni#4219] [Q&A] [answered]
Thank you for a really great job!
For GAN there are generator and discriminator.I would like to ask if it is possible to use NNI for model compression for this group of networks

**Comment by xiaowu0162** [comment, marked as answer]
Hi @yy2016417 , 
Yes. Since most generative adversarial networks use `Linear` and `Conv2d` as their basic building blocks, you can apply NNI pruning and quantization tools to compress them. You can refer to [here](https://github.com/microsoft/nni/tree/master/examples/model_compress) for examples of how to specify a `config_list` and use the tools provided by NNI. Although we don't have an example for compressing GANs, the general process should be very similar.
As a side remark, I think only compressing the generator should be enough for most cases.

**Comment by yy2016417** [comment]
> 
> 
> Hi @yy2016417 ,
> Yes. Since most generative adversarial networks use `Linear` and `Conv2d` as their basic building blocks, you can apply NNI pruning and quantization tools to compress them. You can refer to [here](https://github.com/microsoft/nni/tree/master/examples/model_compress) for examples of how to specify a `config_list` and use the tools provided by NNI. Although we don't have an example for compressing GANs, the general process should be very similar.
> As a side remark, I think only compressing the generator should be enough for most cases.

Thanks for your help !
